src/Agent.Worker/Variables.cs

```diff
@@ -239,38 +239,17 @@ public string Get(string name)
 
         public void Set(string name, string val, bool secret = false)
```

Comment:
Not every variable set using Set is an output variable. Only variables set using task.SetVariable should be marked as Output variables.